src/EditorFeatures/Test/SolutionCrawler/WorkCoordinatorTests.cs

```diff
@@ -871,9 +874,31 @@ End Property
 
                 await WaitWaiterAsync(workspace.ExportProvider);
 
-                // mutate solution
-                workspace.OnSolutionAdded(solution);
+                // block solution cralwer from processing.
+                var globalOperation = workspace.Services.GetService<IGlobalOperationNotificationService>();
+                using (var operation = globalOperation.Start("Block SolutionCrawler"))
+                {
+                    // mutate solution
+                    workspace.OnSolutionAdded(solution);
+
+                    // wait for workspace events to be all processed
+                    var workspaceWaiter = GetListenerProvider(workspace.ExportProvider).GetWaiter(FeatureAttribute.Workspace);
+                    await workspaceWaiter.CreateWaitTask();
+
+                    // now wait for semantic processor to finish
+                    var crawlerListener = (AsynchronousOperationListener)GetListenerProvider(workspace.ExportProvider).GetListener(FeatureAttribute.SolutionCrawler);
+
+                    // first, wait for first work to be queued.
+                    await crawlerListener.CreateWaitTask(tokens => tokens.Any(token => token.Name == "EnqueueWorkItemAsync"));
+
+                    // and then wait them to be processed
+                    await crawlerListener.CreateWaitTask(tokens => tokens.Where(token => token.Tag == workspace).IsEmpty());
+
+                    // let analyzer to process
+                    operation.Done();
```

Comment:
❓ How does this let analyzers proceed?

Reply:
solution crawler will pause processing analyzer if there is pending global operation.